File: frontend/lib/user-segmentation.ts
export type UserSegment = 'CasualViewer' | 'HardcoreFan' | 'DataNerd' | 'GamblerFantasy' | 'SocialViewer';

export interface SegmentationData {
  cyclingLevel: 'new' | 'casual' | 'superfan' | null;
  viewPreference: 'clean' | 'data-rich' | null;
  chatParticipation?: number; // Number of chat messages
  watchTime?: number; // Total watch time in minutes
  points?: number; // Total points
}

/**
 * Determines user segment based on onboarding answers and behavior
 */
export function determineUserSegment(data: SegmentationData): UserSegment {
  // Primary segmentation based on onboarding
  if (data.cyclingLevel === 'new' || data.viewPreference === 'clean') {
    return 'CasualViewer';
  }

  if (data.cyclingLevel === 'superfan' && data.viewPreference === 'data-rich') {
    return 'HardcoreFan';
  }

  if (data.viewPreference === 'data-rich') {
    return 'DataNerd';
  }

  // Secondary segmentation based on behavior (if available)
  if (data.chatParticipation && data.chatParticipation > 100) {
    return 'SocialViewer';
  }

  // Default to casual viewer
  return 'CasualViewer';
}

/**
 * Get segment-specific defaults
 */
export function getSegmentDefaults(segment: UserSegment) {
  switch (segment) {
    case 'CasualViewer':
      return {
        dataMode: 'casual' as const,
        showMinimalUI: true,
        defaultViewMode: 'clean' as const,
      };
    case 'HardcoreFan':
      return {
        dataMode: 'pro' as const,
        showMinimalUI: false,
        defaultViewMode: 'data-rich' as const,
      };
    case 'DataNerd':
      return {
        dataMode: 'pro' as const,
        showMinimalUI: false,
        defaultViewMode: 'data-rich' as const,
      };
    case 'GamblerFantasy':
      return {
        dataMode: 'standard' as const,
        showMinimalUI: false,
        defaultViewMode: 'standard' as const,
      };
    case 'SocialViewer':
      return {
        dataMode: 'standard' as const,
        showMinimalUI: false,
        defaultViewMode: 'standard' as const,
      };
    default:
      return {
        dataMode: 'standard' as const,
        showMinimalUI: false,
        defaultViewMode: 'standard' as const,
      };
  }
}

File: backend/internal/services/user_segmentation.go
package services

// UserSegment represents a user segment type
type UserSegment string

const (
	SegmentCasualViewer  UserSegment = "CasualViewer"
	SegmentHardcoreFan   UserSegment = "HardcoreFan"
	SegmentDataNerd      UserSegment = "DataNerd"
	SegmentGamblerFantasy UserSegment = "GamblerFantasy"
	SegmentSocialViewer  UserSegment = "SocialViewer"
)

// SegmentationData contains data used for user segmentation
type SegmentationData struct {
	CyclingLevel      string // "new", "casual", "superfan"
	ViewPreference    string // "clean", "data-rich"
	ChatParticipation int    // Number of chat messages
	WatchTime         int    // Total watch time in minutes
	Points            int    // Total points
}

// DetermineUserSegment determines the user segment based on onboarding answers and behavior
func DetermineUserSegment(data SegmentationData) UserSegment {
	// Primary segmentation based on onboarding
	if data.CyclingLevel == "new" || data.ViewPreference == "clean" {
		return SegmentCasualViewer
	}

	if data.CyclingLevel == "superfan" && data.ViewPreference == "data-rich" {
		return SegmentHardcoreFan
	}

	if data.ViewPreference == "data-rich" {
		return SegmentDataNerd
	}

	// Secondary segmentation based on behavior
	if data.ChatParticipation > 100 {
		return SegmentSocialViewer
	}

	// Default to casual viewer
	return SegmentCasualViewer
}

// GetSegmentDefaults returns segment-specific defaults
func GetSegmentDefaults(segment UserSegment) map[string]interface{} {
	switch segment {
	case SegmentCasualViewer:
		return map[string]interface{}{
			"data_mode":       "casual",
			"show_minimal_ui": true,
			"default_view_mode": "clean",
		}
	case SegmentHardcoreFan:
		return map[string]interface{}{
			"data_mode":       "pro",
			"show_minimal_ui": false,
			"default_view_mode": "data-rich",
		}
	case SegmentDataNerd:
		return map[string]interface{}{
			"data_mode":       "pro",
			"show_minimal_ui": false,
			"default_view_mode": "data-rich",
		}
	case SegmentGamblerFantasy:
		return map[string]interface{}{
			"data_mode":       "standard",
			"show_minimal_ui": false,
			"default_view_mode": "standard",
		}
	case SegmentSocialViewer:
		return map[string]interface{}{
			"data_mode":       "standard",
			"show_minimal_ui": false,
			"default_view_mode": "standard",
		}
	default:
		return map[string]interface{}{
			"data_mode":       "standard",
			"show_minimal_ui": false,
			"default_view_mode": "standard",
		}
	}
}

